src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleExtensionUsage.java

```diff
@@ -130,10 +131,26 @@ public final boolean getHasNonDevUseExtension() {
     return getProxies().stream().anyMatch(p -> !p.isDevDependency());
   }
 
+  @GenerateTypeAdapter
+  public record RepoOverride(
+      // The apparent name of the overriding repo in the root module.
+      String overridingRepoName,
+      // Whether this override should apply to an existing repo.
+      boolean mustExist,
+      Location location) {}
+
+  /**
+   * Maps repo names local to the extension to repo they are overridden by.
+   *
+   * <p>This is only non-empty for root module usages and repos that override other repos are not
```

Comment:
The "and repos that override ..." part confuses me. Why is it worth pointing out here?

Reply:
sorry, I'm still confused.

1. "This is only non-empty for root module usages" stands by itself, right? If so, maybe add a comma (or even a period) before "and repos that ...".
2. Why does the fact that overriding repos cannot be overridden mean "overrides do need to applied recursively"? What does applying an override recursively even mean?